python/src/etos_api/routers/v0/router.py

```diff
@@ -20,57 +20,76 @@
+
+ETOSv0 = FastAPI(
+    title="ETOS",
+    version="v0",
+    summary="API endpoints for ETOS v0 - I.e. the version before versions",
+    root_path_in_servers=False,
+)
 TRACER = trace.get_tracer("etos_api.routers.etos.router")
 LOGGER = logging.getLogger(__name__)
 logging.getLogger("pika").setLevel(logging.WARNING)
 
 
-async def download_suite(test_suite_url: str) -> dict:
-    """Attempt to download suite.
+@ETOSv0.post("/etos", tags=["etos"], response_model=StartEtosResponse)
+async def start_etos(etos: StartEtosRequest):
+    """Start ETOS execution on post.
 
-    :param test_suite_url: URL to test suite to download.
-    :return: Downloaded test suite as JSON.
+    :param etos: ETOS pydantic model.
+    :type etos: :obj:`etos_api.routers.etos.schemas.StartEtosRequest`
+    :return: JSON dictionary with response.
+    :rtype: dict
     """
-    try:
-        suite = requests.get(test_suite_url, timeout=60)
-        suite.raise_for_status()
-    except Exception as exception:  # pylint:disable=broad-except
-        raise AssertionError(f"Unable to download suite from {test_suite_url}") from exception
-    return suite.json()
+    with TRACER.start_as_current_span("start-etos") as span:
+        return await _start(etos, span)
 
 
-async def validate_suite(test_suite_url: str) -> None:
-    """Validate the ETOS test suite through the SuiteValidator.
+@ETOSv0.delete("/etos/{suite_id}", tags=["etos"], response_model=AbortEtosResponse)
+async def abort_etos(suite_id: str):
+    """Abort ETOS execution on delete.
 
-    :param test_suite_url: The URL to the test suite to validate.
+    :param suite_id: ETOS suite id
+    :type suite_id: str
+    :return: JSON dictionary with response.
+    :rtype: dict
     """
-    span = trace.get_current_span()
+    with TRACER.start_as_current_span("abort-etos"):
+        return await _abort(suite_id)
 
-    try:
-        test_suite = await download_suite(test_suite_url)
-        await SuiteValidator().validate(test_suite)
-    except AssertionError as exception:
-        LOGGER.error("Test suite validation failed!")
-        LOGGER.error(exception)
-        span.add_event("Test suite validation failed")
-        raise HTTPException(
-            status_code=400, detail=f"Test suite validation failed. {exception}"
-        ) from exception
+
+@ETOSv0.get("/ping", tags=["etos"], status_code=204)
+async def ping():
+    """Ping the ETOS service in order to check if it is up and running.
+
+    :return: HTTP 204 response.
+    :rtype: :obj:`starlette.responses.Response`
+    """
+    return Response(status_code=204)
+
+
+@ETOSv0.get("/selftest/ping")
+async def oldping():
+    """Ping the ETOS service in order to check if it is up and running.
+
+    This is deprecated in favor of `/api/etos/ping`. Implementing here
+    for backward compatibility. In newer API versions this shall not
+    exist.
+    """
+    LOGGER.warning("DEPRECATED request to selftest/ping received!")
```

Comment:
No risk for too much logging?

Reply:
I don't think it will be that bad, considering we already have the `INFO:     127.0.0.1:12345 - "GET /selftest/ping HTTP/1.1" 204 No Content` printed every second.
But it's a nice indicator that we should update the ETOS Client to utilize the new endpoint instead (which is something that I will work on soon)

Reply:
@fredjn Do you see any issues? Should be fine?

Reply:
The only issue I'va had with the "I'm fine, I'm fine, ..."-INFO-spamming is that it sometimes forced me to filter it out since i clutters the rest of the logs sometimes. However this warning might be useful to track down clients using old versions of API(s), if we (ever) get OTEL log correlation working it should be fairly trivial, and thus the WARN has a nice overall value.